dashboard/scripts/update_under_13_students.rb

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env ruby
+# Update all students under age 13, or with age not set, to have sharing_disabled = true.
+# sharing_disabled is contained in properties of the User model
+require_relative('../config/environment')
+
+puts "Starting to batch update all students under 13"
+
+min_birthday = Date.today - 13.years
+users = User.where('birthday IS NULL OR birthday > ?', min_birthday)
+users.find_each do |user|
+  user.update! sharing_disabled: true
+end
```

Comment:
I'm not convinced that updating individual records is better than doing it in batches, because it leads to many more database writes. however this seems like it will work, and I'm not familiar with the syntax for updating in batches without locking the table for a transaction across all the items in the batch.

since this might take a while to run, you might find it useful to print something out after every 100,000 users are updated.

Reply:
Since the field being written is in serialized_attributes, each row will potentially have a different value set. Otherwise we could just rely on SQL to update all rows at once. Given that each row will need to be individually inspected, I don't know a good way to batch the writes. Containing them in a transaction adds additional locks, but it would still be multiple sql update statements.